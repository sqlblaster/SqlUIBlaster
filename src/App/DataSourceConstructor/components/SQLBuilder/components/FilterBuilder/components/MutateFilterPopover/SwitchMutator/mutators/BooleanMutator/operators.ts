enum BooleanOperatorEnum {
  'Is true',
  'Is false',
  'Is empty',
  'Is not empty'
}

export type BooleanOperator = keyof typeof BooleanOperatorEnum;

export const booleanOperators: BooleanOperator[] = [
  'Is true',
  'Is false',
  'Is empty',
  'Is not empty'
];
