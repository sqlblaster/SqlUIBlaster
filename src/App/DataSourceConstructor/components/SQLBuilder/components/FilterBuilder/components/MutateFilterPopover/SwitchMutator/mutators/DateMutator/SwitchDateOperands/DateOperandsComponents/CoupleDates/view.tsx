import { withStyles } from '@material-ui/core';
import * as React from 'react';
import { DatePickerByOperandsType } from '../../DatePickerByOperandsType';
import { CoupleDateViewProps } from './props';
import { coupleDateStyles } from './styles';

export const CoupleDate = withStyles(coupleDateStyles)((({
  operatorView,
  operands,
  pickedVariables,
  classes,
  handleVariablePick,
  handleOperandsChange
}) => {
  return (
    <>
      <div className={classes.root}>
        {operatorView}
        <div className={classes.dates}>
          <div className={classes.date} data-testid='between-operand-1'>
            <DatePickerByOperandsType
              operands={operands[0]}
              onOperandsChange={handleOperandsChange(0)}
              variable={pickedVariables[0]}
              onVariablePicked={handleVariablePick(0)}
              limits={
                operands[1].dateType === 'Fixed date'
                  ? {
                      maxDate: operands[1].date,
                      maxDateMessage:
                        'Couldn\'t set time, because selected date is later than second date'
                    }
                  : undefined
              }
            />
          </div>
          <div className={classes.date} data-testid='between-operand-2'>
            <DatePickerByOperandsType
              operands={operands[1]}
              onOperandsChange={handleOperandsChange(1)}
              variable={pickedVariables[1]}
              onVariablePicked={handleVariablePick(1)}
              limits={
                operands[0].dateType === 'Fixed date'
                  ? {
                      minDate: operands[0].date,
                      minDateMessage:
                        'Couldn\'t set time, because selected date is earlier than first date'
                    }
                  : undefined
              }
            />
          </div>
        </div>
      </div>
    </>
  );
}) as React.FC<CoupleDateViewProps>);
