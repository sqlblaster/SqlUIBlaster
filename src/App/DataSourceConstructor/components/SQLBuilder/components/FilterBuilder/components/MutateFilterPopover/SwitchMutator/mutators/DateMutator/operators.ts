enum DateOperatorEnum {
  'Previous',
  'Next',
  'Before',
  'After',
  'On',
  'Between',
  'Is empty',
  'Is not empty'
}

export type DateOperator = keyof typeof DateOperatorEnum;

export const dateOperators: DateOperator[] = [
  'Previous',
  'Next',
  'Before',
  'After',
  'On',
  'Between',
  'Is empty',
  'Is not empty'
];
