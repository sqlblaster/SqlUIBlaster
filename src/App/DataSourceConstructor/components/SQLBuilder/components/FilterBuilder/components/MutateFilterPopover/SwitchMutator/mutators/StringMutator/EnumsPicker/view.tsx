import { FormControl, MenuItem, Select, withStyles } from '@material-ui/core';
import * as React from 'react';
import { VariableItem } from '../../../VariableItem';
import { VariablesPicker } from '../../../VariablesPicker';
import { EnumsPickerViewProps } from './props';
import { enumsPickerStyles } from './styles';

export const EnumsPicker = withStyles(enumsPickerStyles)((({
  selectedEnum,
  selectedVariable,
  enums,
  classes,
  selectIsOpen,
  filterVariables,
  handleEnumChange,
  handleVariablePick,
  handleSelectOpen,
  handleSelectClose,
  handleVariableItemClick
}) => {
  return (
    <FormControl className={classes['form-control']}>
      <Select
        value={selectedVariable ? '' : selectedEnum}
        onChange={handleEnumChange}
        open={selectIsOpen}
        onOpen={handleSelectOpen}
        onClose={handleSelectClose}
        classes={{ select: classes.select }}
        endAdornment={
          <VariablesPicker
            onVariablePicked={handleVariablePick}
            filter={filterVariables}
          />
        }
        startAdornment={
          <>
            {selectedVariable && (
              <VariableItem
                className={classes['variable-item']}
                variable={selectedVariable}
                onVariableClick={handleVariableItemClick}
              />
            )}
          </>
        }
      >
        {enums.map(enumuration => (
          <MenuItem key={enumuration} value={enumuration}>
            {enumuration}
          </MenuItem>
        ))}
      </Select>
    </FormControl>
  );
}) as React.FC<EnumsPickerViewProps>);
