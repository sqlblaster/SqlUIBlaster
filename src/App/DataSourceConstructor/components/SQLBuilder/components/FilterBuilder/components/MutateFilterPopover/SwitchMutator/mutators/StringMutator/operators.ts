enum StringOperatorEnum {
  'Is',
  'Is not',
  'Contains',
  'Does not contain',
  'Starts with',
  'Ends with',
  'Is empty',
  'Is not empty'
}

export type StringOperator = keyof typeof StringOperatorEnum;

export const stringOperators: StringOperator[] = [
  'Is',
  'Is not',
  'Contains',
  'Does not contain',
  'Starts with',
  'Ends with',
  'Is empty',
  'Is not empty'
];
