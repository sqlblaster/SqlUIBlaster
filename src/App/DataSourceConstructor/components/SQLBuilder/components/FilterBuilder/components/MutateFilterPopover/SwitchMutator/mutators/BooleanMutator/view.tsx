import { withStyles } from '@material-ui/core';
import * as React from 'react';
import { CheckedSelect } from '../../CheckedSelect';
import { BooleanOperator, booleanOperators } from './operators';
import { BooleanMutatorViewProps } from './props';
import { booleanMutatorStyles } from './styles';

export const BooleanMutator = withStyles(booleanMutatorStyles)(
  class extends React.Component<BooleanMutatorViewProps> {
    public render() {
      const {
        operator,
        handleOperatorSelect,
        titleComponent,
        classes,
        submitMutationComponent
      } = this.props;

      return (
        <div className={classes.root}>
          <div
            className={ `${classes['title-and-operator']} ${
              classes['common-padding']
            }` }
          >
            <div className={classes.title}>{titleComponent}</div>
            <div className={classes.operator}>
              <CheckedSelect<BooleanOperator>
                selectedItem={operator}
                items={booleanOperators}
                onSelect={handleOperatorSelect}
              />
            </div>
          </div>
          <hr className={classes.hr} />
          <div
            className={ `${classes['mutator-action']} ${
              classes['common-padding']
            }` }
          >
            {submitMutationComponent}
          </div>
        </div>
      );
    }
  }
);
