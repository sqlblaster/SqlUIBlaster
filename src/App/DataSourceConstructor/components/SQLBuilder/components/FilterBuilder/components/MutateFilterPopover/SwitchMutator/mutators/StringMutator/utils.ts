import { StringOperator } from './operators';

export const hasCaseSensivity = (operator: StringOperator): boolean => {
  switch (operator) {
    case 'Contains':
    case 'Does not contain':
    case 'Starts with':
    case 'Ends with':
      return true;
    default:
      return false;
  }
};
