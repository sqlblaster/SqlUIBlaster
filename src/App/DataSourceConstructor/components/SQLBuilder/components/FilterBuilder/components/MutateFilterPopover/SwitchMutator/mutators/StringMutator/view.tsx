import {
  Checkbox,
  FormControlLabel,
  Input,
  withStyles
} from '@material-ui/core';
import * as React from 'react';

import { VariableType } from 'src/schema';
import { CheckedSelect } from '../../CheckedSelect';
import { VariableItem } from '../../VariableItem';
import { VariablesPicker } from '../../VariablesPicker';
import { EnumsPicker } from './EnumsPicker';
import { StringOperator, stringOperators } from './operators';
import { StringMutatorViewProps } from './props';
import { stringMutatorStyles } from './styles';
import { hasCaseSensivity } from './utils';

export const stringMutatorPlaceholder = 'type desired text';
export const caseSensitivityText = 'Case sensitive';

export const StringMutator = withStyles(stringMutatorStyles)(
  class extends React.Component<StringMutatorViewProps> {
    public render() {
      const {
        operator,
        operands,
        caseSensitive,
        enums,
        column,
        pickedVariables,
        titleComponent,
        classes,
        submitMutationComponent,
        handleOperatorSelect: handleSelect,
        handleOperandChange,
        handleTextFieldChange,
        handleCaseSensitivityChange,
        handleVariableSelect,
        handleVariableClick
      } = this.props;

      return (
        <div className={classes.root}>
          <div
            className={`${classes['title-and-operator']} ${classes['common-padding']}`}
          >
            <div className={classes.title}>{titleComponent}</div>
            <div className={classes.operator}>
              <CheckedSelect<StringOperator>
                selectedItem={operator}
                items={stringOperators}
                onSelect={handleSelect}
              />
            </div>
          </div>
          <hr className={classes.hr} />
          {operator !== 'Is empty' && operator !== 'Is not empty' && (
            <>
              {
                <div
                  className={`${classes.operands} ${classes['common-padding']}`}
                >
                  {enums && enums.length ? (
                    <EnumsPicker
                      selectedEnum={operands}
                      selectedVariable={
                        pickedVariables instanceof Array
                          ? null
                          : pickedVariables
                      }
                      enums={enums}
                      column={column}
                      onEnumSelect={handleOperandChange}
                      onVariableSelect={handleVariableSelect}
                    />
                  ) : (
                    <Input
                      fullWidth
                      autoComplete='off'
                      value={operands}
                      placeholder={
                        pickedVariables ? '' : stringMutatorPlaceholder
                      }
                      disabled={!!pickedVariables}
                      startAdornment={
                        pickedVariables &&
                        !(pickedVariables instanceof Array) && (
                          <VariableItem
                            className={classes['variable-item']}
                            variable={pickedVariables}
                            onVariableClick={handleVariableClick}
                          />
                        )
                      }
                      endAdornment={
                        <VariablesPicker
                          type={VariableType.String}
                          onVariablePicked={handleVariableSelect}
                        />
                      }
                      onChange={handleTextFieldChange}
                    />
                  )}
                </div>
              }
            </>
          )}
          <hr className={classes.hr} />
          <div
            className={`${classes['mutator-action']} ${classes['common-padding']}`}
          >
            {hasCaseSensivity(operator) ? (
              <FormControlLabel
                control={
                  <Checkbox
                    id='case-sensitive'
                    checked={caseSensitive}
                    onChange={handleCaseSensitivityChange}
                    color='primary'
                    className={classes.CheckBox}
                  />
                }
                htmlFor='case-sensitive'
                label={caseSensitivityText}
                classes={{
                  root: classes.FormControlLabel,
                  label: classes.label
                }}
              />
            ) : (
              <span />
            )}
            {submitMutationComponent}
          </div>
        </div>
      );
    }
  }
);
