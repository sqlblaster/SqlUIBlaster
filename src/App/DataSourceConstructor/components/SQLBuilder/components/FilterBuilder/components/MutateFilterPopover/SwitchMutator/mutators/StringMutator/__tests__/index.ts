import {
  fireEvent,
  getAllByText as getAllByTextOn,
  getByDisplayValue as getByDisplayValueOn,
  getByText as getByTextOn,
  wait
} from 'react-testing-library';
import {
  expandMoreIconPlaceholder,
  shareIconPlaceholder,
  variableIconPlaceholder
} from 'src/__tests__/utils/mock-placeholders';
import {
  getGeneratedQuery,
  render
} from 'src/App/DataSourceConstructor/components/SQLBuilder/__tests__/render.utils';
import {
  clickAddTheFilter,
  clickUpdateTheFilter
} from 'src/App/DataSourceConstructor/components/SQLBuilder/components/FilterBuilder/__tests__/utils/filter-utils';
import { selectATable } from 'src/App/DataSourceConstructor/components/SQLBuilder/components/TableSelector/__tests__/utils/select-table';
import { tablesMock } from 'src/App/DataSourceConstructor/redux/tables/tables.mock';
import {
  getAllByText$,
  getByText$,
  regex
} from 'src/App/DataSourceConstructor/test-utils/utils';
import { clickAddAFilter } from '../../../../../AddFilter/__tests__/utils/click-add-a-filter';
import { addFilterText, mutatorPanelId } from '../../../../view';
import { StringOperator, stringOperators } from '../operators';
import { caseSensitivityText, stringMutatorPlaceholder } from '../view';
import { enumVariableMock } from './utils/enum-variables-mock';

describe(`String operations`, () => {
  it(`should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  `, async () => {
    const {
      rerender,
      getByText,
      queryByText,
      getByTestId,
      getByLabelText,
      getByPlaceholderText,
      componentNode,
      debug
    } = render();

    const columns = selectATable();
    const foreignColumnName = columns[5].name;
    const innerField = tablesMock[2].columns[1].name;

    let clickOnFilter = () => {
      fireEvent.click(getByText(regex(innerField)));
    };

    const checkFilterDescription = (operator: StringOperator) => {
      expect(
        getByText$(
          componentNode,
          new RegExp(`^${foreignColumnName}.*${innerField}.+$`, 'i')
        ).textContent
      ).toMatchSnapshot(`filter description for '${operator}' date operator`);
    };

    const checkOperation = (
      nextOperator: StringOperator,
      prevOperator: StringOperator,
      caseSensitive: boolean = false,
      doBeforeSave: () => void = () => null
    ) => {
      clickOnFilter();
      fireEvent.click(getByText(new RegExp(`^${prevOperator}.*$`)));
      fireEvent.click(getByText(new RegExp(`^.*${nextOperator}$`)));
      doBeforeSave();
      clickUpdateTheFilter();

      checkFilterDescription(nextOperator);

      expect(getGeneratedQuery()).toMatchSnapshot(
        `${
          caseSensitive ? 'case sensitive ' : ''
        }'${nextOperator}' string operator`
      );
    };

    // default string filter
    clickAddAFilter();
    fireEvent.click(
      getByTextOn(
        getByText$(
          componentNode,
          regex(`${shareIconPlaceholder}.*${foreignColumnName}.*`)
        ),
        regex(expandMoreIconPlaceholder)
      )
    );
    fireEvent.click(getByText(regex(innerField)));

    const textField = getByPlaceholderText(
      regex(stringMutatorPlaceholder)
    ) as HTMLInputElement;
    textField.defaultValue = 'Candy';
    fireEvent.input(textField);
    clickAddTheFilter();

    checkFilterDescription('Is');

    expect(getGeneratedQuery()).toMatchSnapshot(`default string operator`);

    // Check all operators
    for (let i = 1; i < stringOperators.length; i++) {
      const nextOperator = stringOperators[i];
      const prevOperator = stringOperators[i - 1];
      checkOperation(nextOperator, prevOperator);
    }

    // Check case sensitive operators
    const caseSensitiveOperators = stringOperators.slice(2, 6);
    for (let i = 0; i < caseSensitiveOperators.length; i++) {
      const prevOperator: StringOperator =
        i === 0 ? 'Is not empty' : caseSensitiveOperators[i - 1];
      const nextOperator = caseSensitiveOperators[i];
      checkOperation(
        nextOperator,
        prevOperator,
        true,
        i !== 0
          ? () => null
          : () => {
              fireEvent.click(getByLabelText(caseSensitivityText));
            }
      );
    }

    // Check enum string operation
    const columnName = columns[1].name;
    clickOnFilter = () => {
      fireEvent.click(getByText(columnName));
    };
    const enums = columns[1].enums as string[];
    fireEvent.click(getByText(/clear/i));
    clickAddAFilter();
    fireEvent.click(getByText(regex(columnName)));
    clickAddTheFilter();

    expect(getGeneratedQuery()).toMatchSnapshot(`Enum string operation`);

    clickOnFilter();
    fireEvent.click(getByTextOn(getByTestId(mutatorPanelId), enums[0]));
    fireEvent.click(getByTextOn(document.body, enums[1]));
    clickUpdateTheFilter();

    expect(getGeneratedQuery()).toMatchSnapshot(
      `Enum string operation with second enum's value`
    );

    // Check the variable pick
    const validVariable = enumVariableMock[1].name;
    rerender({ variables: enumVariableMock });
    clickOnFilter();
    fireEvent.click(getByText(variableIconPlaceholder));

    // The rest of variable filtered properly
    expect(queryByText(enumVariableMock[0].name)).toBeNull();
    expect(queryByText(enumVariableMock[2].name)).toBeNull();

    fireEvent.click(getByText(validVariable));

    expect(getByText(validVariable).textContent).toMatchSnapshot(
      'valid variable is selected'
    );

    clickUpdateTheFilter();

    expect(
      getByText$(
        componentNode,
        regex(`${columnName}.*${stringOperators[0]}.*${validVariable}$`)
      ).textContent
    ).toMatchSnapshot('valid variable is drawn correctly');

    expect(getGeneratedQuery()).toMatchSnapshot(
      `Variable picked after the enum value pick`
    );

    // Check picking enum value after variable pick
    clickOnFilter();
    fireEvent.click(
      getByTextOn(getByTestId(mutatorPanelId), regex(validVariable))
    );
    fireEvent.click(getByTextOn(document.body, enums[1]));
    clickUpdateTheFilter();

    expect(getGeneratedQuery()).toMatchSnapshot(
      `Enum value picked after variable pick`
    );
  });
});
