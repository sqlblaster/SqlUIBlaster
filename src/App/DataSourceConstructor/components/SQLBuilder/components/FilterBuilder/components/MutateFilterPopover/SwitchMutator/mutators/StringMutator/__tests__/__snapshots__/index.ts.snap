// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : 'Contains' string operator 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
  LEFT JOIN \\"products\\" \\"products1\\" ON \\"orders1\\".\\"ProductID\\" = \\"products1\\".\\"ID\\"
WHERE
  lower(\\"products1\\".\\"Product name\\"::text)
  LIKE concat('%', lower('Candy'), '%')
LIMIT 2000
"
`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : 'Does not contain' string operator 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
  LEFT JOIN \\"products\\" \\"products1\\" ON \\"orders1\\".\\"ProductID\\" = \\"products1\\".\\"ID\\"
WHERE
  lower(\\"products1\\".\\"Product name\\"::text)
  NOT LIKE concat('%', lower('Candy'), '%')
LIMIT 2000
"
`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : 'Ends with' string operator 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
  LEFT JOIN \\"products\\" \\"products1\\" ON \\"orders1\\".\\"ProductID\\" = \\"products1\\".\\"ID\\"
WHERE
  lower(\\"products1\\".\\"Product name\\"::text)
  LIKE concat('%', lower('Candy'))
LIMIT 2000
"
`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : 'Is empty' string operator 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
  LEFT JOIN \\"products\\" \\"products1\\" ON \\"orders1\\".\\"ProductID\\" = \\"products1\\".\\"ID\\"
WHERE
  \\"products1\\".\\"Product name\\" IS NULL
LIMIT 2000
"
`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : 'Is not empty' string operator 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
  LEFT JOIN \\"products\\" \\"products1\\" ON \\"orders1\\".\\"ProductID\\" = \\"products1\\".\\"ID\\"
WHERE
  \\"products1\\".\\"Product name\\" IS NOT NULL
LIMIT 2000
"
`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : 'Is not' string operator 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
  LEFT JOIN \\"products\\" \\"products1\\" ON \\"orders1\\".\\"ProductID\\" = \\"products1\\".\\"ID\\"
WHERE
  \\"products1\\".\\"Product name\\"::text <> 'Candy'
LIMIT 2000
"
`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : 'Starts with' string operator 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
  LEFT JOIN \\"products\\" \\"products1\\" ON \\"orders1\\".\\"ProductID\\" = \\"products1\\".\\"ID\\"
WHERE
  lower(\\"products1\\".\\"Product name\\"::text)
  LIKE concat(lower('Candy'), '%')
LIMIT 2000
"
`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : Enum string operation 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Details\\"::text = 'Detail1'
LIMIT 2000
"
`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : Enum string operation with second enum's value 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Details\\"::text = 'Detail2'
LIMIT 2000
"
`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : Enum value picked after variable pick 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Details\\"::text = 'Detail2'
LIMIT 2000
"
`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : Variable picked after the enum value pick 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
WHERE
  \\"orders1\\".\\"Details\\"::text = $2::text
LIMIT 2000
"
`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : case sensitive 'Contains' string operator 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
  LEFT JOIN \\"products\\" \\"products1\\" ON \\"orders1\\".\\"ProductID\\" = \\"products1\\".\\"ID\\"
WHERE
  \\"products1\\".\\"Product name\\"::text LIKE concat('%', 'Candy', '%')
LIMIT 2000
"
`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : case sensitive 'Does not contain' string operator 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
  LEFT JOIN \\"products\\" \\"products1\\" ON \\"orders1\\".\\"ProductID\\" = \\"products1\\".\\"ID\\"
WHERE
  \\"products1\\".\\"Product name\\"::text NOT LIKE concat('%', 'Candy', '%')
LIMIT 2000
"
`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : case sensitive 'Ends with' string operator 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
  LEFT JOIN \\"products\\" \\"products1\\" ON \\"orders1\\".\\"ProductID\\" = \\"products1\\".\\"ID\\"
WHERE
  \\"products1\\".\\"Product name\\"::text LIKE concat('%', 'Candy')
LIMIT 2000
"
`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : case sensitive 'Starts with' string operator 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
  LEFT JOIN \\"products\\" \\"products1\\" ON \\"orders1\\".\\"ProductID\\" = \\"products1\\".\\"ID\\"
WHERE
  \\"products1\\".\\"Product name\\"::text LIKE concat('Candy', '%')
LIMIT 2000
"
`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : default string operator 1`] = `
"SELECT
  orders1.*
FROM
  \\"orders\\" \\"orders1\\"
  LEFT JOIN \\"products\\" \\"products1\\" ON \\"orders1\\".\\"ProductID\\" = \\"products1\\".\\"ID\\"
WHERE
  \\"products1\\".\\"Product name\\"::text = 'Candy'
LIMIT 2000
"
`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : filter description for 'Contains' date operator 1`] = `"ProductIDShareIconProduct namecontainsCandy"`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : filter description for 'Contains' date operator 2`] = `"ProductIDShareIconProduct namecontainsCandy"`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : filter description for 'Does not contain' date operator 1`] = `"ProductIDShareIconProduct namedoes not containCandy"`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : filter description for 'Does not contain' date operator 2`] = `"ProductIDShareIconProduct namedoes not containCandy"`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : filter description for 'Ends with' date operator 1`] = `"ProductIDShareIconProduct nameends withCandy"`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : filter description for 'Ends with' date operator 2`] = `"ProductIDShareIconProduct nameends withCandy"`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : filter description for 'Is empty' date operator 1`] = `"ProductIDShareIconProduct nameis empty"`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : filter description for 'Is not empty' date operator 1`] = `"ProductIDShareIconProduct nameis not empty"`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : filter description for 'Is not' date operator 1`] = `"ProductIDShareIconProduct nameis notCandy"`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : filter description for 'Is' date operator 1`] = `"ProductIDShareIconProduct nameisCandy"`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : filter description for 'Starts with' date operator 1`] = `"ProductIDShareIconProduct namestarts withCandy"`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : filter description for 'Starts with' date operator 2`] = `"ProductIDShareIconProduct namestarts withCandy"`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : valid variable is drawn correctly 1`] = `"DetailsisVariableIconOrder details"`;

exports[`String operations should render proper sql query for:
      - primary fields,
      - enum fields,
      - case insensitive cases,
      should check the proper filter description render,
      should check the variable pick and from variable to enum pick
  : valid variable is selected 1`] = `"VariableIconOrder details"`;
